allofplos/makedb.py

```diff
@@ -105,26 +104,32 @@ class PLOSArticle(BaseModel):
     word_count = IntegerField()
     JATS_type = ForeignKeyField(JATSType, related_name='jats')
 
+class SubjectsPLOSArticle(BaseModel):
+    subject = ForeignKeyField(Subjects)
+    article = ForeignKeyField(PLOSArticle)
+
 class CoAuthorPLOSArticle(BaseModel):
     corr_author = ForeignKeyField(CorrespondingAuthor)
     article = ForeignKeyField(PLOSArticle)
 
 db.connect()
-db.create_tables([Journal, PLOSArticle, ArticleType,
-                      CoAuthorPLOSArticle, CorrespondingAuthor,
-                      JATSType, Affiliations, Country])
+db.create_tables([Journal, PLOSArticle, ArticleType, CoAuthorPLOSArticle,
+                  CorrespondingAuthor, JATSType, Affiliations, Country,
+                  SubjectsPLOSArticle, Subjects])
+
 
 if args.starter:
-    allfiles = os.listdir('starter_corpus')
+    corpus = Corpus(starterdir)
+    allfiles = corpus.iter_files
```

Comment:
I think the naming here is a little confusing since it follows a different usage pattern in the three cases. 

is `args.starter` `None` if it is not included?  if so, couldn't you do

```python
corpus_dir = starterdir if args.starter else None
allfiles = Corpus(corpus_dir).iter_files
files = iter(random.sample(list(corpus.iter_files, args.random))) if args.random else allfiles

for file_ in tqdm(files):
⋮
```

with the possibility of changing the second ternary to a full conditional block if its too long for your taste. 

but that would make it more concise and would allow combining a random sample with starterdir in one step.

Reply:
I'll just make a PR against your branch to change this

Reply:
it is OK for me, personally I find easy to read the named and nested option, but I can handle any version, so please go ahead.

Reply:
It's that it's not allfiles when you're sampling… and while I was thinking about it I realised that the Corpus creation didn't need to be its own step since you weren't using the Corpus for anything else.